### PR TITLE
Add examples of serialization utilies for Java (#31)

### PR DESCRIPTION
**Pull request by kozub**
*Issue #, if available:*
#31
*Description of changes:*

Adding examples of serialization utilities for Java base on docs.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: java/SerializationUtilities/Function/src/main/java/org/demo/serialization/APIGatewayRequestDeserializationFunction.java [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    private final static Logger LOGGER = LogManager.getLogger();
+    private static final Map<String, String> HEADERS = Map.of(
+        "Content-Type", "application/json",
+        "X-Custom-Header", "application/json");
+
+    public APIGatewayProxyResponseEvent handleRequest(APIGatewayProxyRequestEvent event, Context context) {
+
+        Product product = extractDataFrom(event).as(Product.class);
+        LOGGER.info("\n=============== Deserialized request body: ===============");
+        LOGGER.info("body={}\n", product);
```
Comment (jeromevdl) [inline java/SerializationUtilities/Function/src/main/java/org/demo/serialization/APIGatewayRequestDeserializationFunction.java:26]:
```suggestion
        LOGGER.info("product={}\n", product);
```

File: java/SerializationUtilities/Function/src/main/java/org/demo/serialization/SQSEventDeserializationFunction.java [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+public class SQSEventDeserializationFunction implements RequestHandler<SQSEvent, String> {
+
+    private final static Logger LOGGER = LogManager.getLogger();
+
+    public String handleRequest(SQSEvent event, Context context) {
+        List<Product> products = extractDataFrom(event).asListOf(Product.class);
+
+        LOGGER.info("\n=============== Deserialized events: ===============");
+        LOGGER.info("events={}\n", products);
```
Comment (jeromevdl) [inline java/SerializationUtilities/Function/src/main/java/org/demo/serialization/SQSEventDeserializationFunction.java:22]:
```suggestion
        LOGGER.info("products={}\n", products);
```

File: java/SerializationUtilities/README.md [hunk 3]
```diff
@@ -0,0 +1,21 @@
+# Parameters
```
Comment (jeromevdl) [inline java/SerializationUtilities/README.md:1]:
```suggestion
# Deserialization
```

File: java/SerializationUtilities/events/APIGatewayEvent.json [hunk 4]
```diff
@@ -0,0 +1,63 @@
+{
```
Comment (jeromevdl) [inline java/SerializationUtilities/events/APIGatewayEvent.json:1]: Is this file actually used?

File: java/SerializationUtilities/events/SQSEvent.json [hunk 5]
```diff
@@ -0,0 +1,39 @@
+{
```
Comment (jeromevdl) [inline java/SerializationUtilities/events/SQSEvent.json:1]: Is this file actually used?

File: java/SerializationUtilities/template.yaml [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+Globals:
+  Function:
+    Timeout: 20
+    Runtime: java11
+    MemorySize: 512
+    Tracing: Active
+
+
+Resources:
+  APIGatewayDes erializationFunction:
```
Comment (jeromevdl) [inline java/SerializationUtilities/template.yaml:15]:
```suggestion
  APIGatewayDeserializationFunction:
```
there's a space in the name

**Review by jeromevdl — CHANGES_REQUESTED** [review]
Few minor things to review, otherwise ok.

File: java/SerializationUtilities/template.yaml [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+            MaximumBatchingWindowInSeconds: 30
+
+
+Outputs:
+  Api:
+    Description: "API Gateway endpoint URL for Prod stage for Serialization function"
+    Value: !Sub "https://${ServerlessRestApi}.execute-api.${AWS::Region}.amazonaws.com/Prod/product/"
+  Function:
+    Description: "Serialization Lambda Function ARN"
+    Value: !GetAtt APIGatewayDeerializationFunction.Arn
```
Comment (orozcoadrian) [inline java/SerializationUtilities/template.yaml:60]:
```suggestion
    Value: !GetAtt APIGatewayDeserializationFunction.Arn
```

File: java/SerializationUtilities/Function/pom.xml [hunk 8]
```diff
@@ -0,0 +1,69 @@
...
+        <dependency>
+            <groupId>org.junit.jupiter</groupId>
+            <artifactId>junit-jupiter-engine</artifactId>
+            <version>5.8.2</version>
+            <scope>test</scope>
+        </dependency>
+    </dependencies>
+
+    <build>
+        <plugins>
```
Comment (orozcoadrian) [inline java/SerializationUtilities/Function/pom.xml:60]: Do you need to add the `aspectj-maven-plugin` configuration? I thought it was required. Afaik it'll seem to log correctly without it but you may see issues with log variables not getting cleared across lambda invocations.

File: java/SerializationUtilities/events/APIGatewayEvent.json (hunk 4 above)
Comment (kozub) [reply java/SerializationUtilities/events/APIGatewayEvent.json:1]: Right now it is not used by SAM template. I've added it for local testing purpose. Let me know if it does not make any sense - I will remove it. 

As far as I understand events from other examples (e.g. for Validation) are also not used - am I right?

Comment (jeromevdl) [reply java/SerializationUtilities/events/APIGatewayEvent.json:1]: OK, no problem. We can keep them.

File: java/SerializationUtilities/Function/pom.xml (hunk 8 above)
Comment (jeromevdl) [reply java/SerializationUtilities/Function/pom.xml:60]: It is useful when using power tools annotations. @Logger, @Idempotent,...  which is not the case here. We can remove it.

Comment (kozub) [reply java/SerializationUtilities/Function/pom.xml:60]: As far as I understand  `aspectj-maven-plugin` is used in Powertools for annotation processing.  Serialization utils doesn't use any annotation (it's just a library that we call directly from the code), so support for aspects is not required to demonstrate this feature. Please correct me if I do not understand it correctly.

**Comment by kozub** [comment]
Thanks @jeromevdl  and @orozcoadrian for your suggestions. I've applied changes that you suggested. Please let me know if there are any additional changes required.

**Pull request merged**